crates/cibse_guide_e/src/chapter_6/equation_6_58/openfire_runner.rs

```diff
@@ -0,0 +1,214 @@
+            r.symbol(),
+            a_t.symbol(),
+            a_o.symbol(),
+            h_o.symbol(),
+            w.symbol(),
+            d.symbol(),
+        )));
+
+        Form::new(vec![step_1])
+    }
+    fn parameters(&self) -> Parameters {
+        let mut params = Parameters::new();
+
+        let r = ParamBuilder::float(SYMBOLS.r)
+            .name("Rate of fuel combustion")
+            .units("kg/s")
+            .build();
+
+        let a_t = ParamBuilder::float(SYMBOLS.a_t)
+            .name("Area of enclosing walls")
+            .units("m^2")
+            .min_exclusive(0.0)
+            .required()
+            .build();
+
+        let a_o = ParamBuilder::float(SYMBOLS.a_o)
+            .name("Area of the opening")
+            .units("m^2")
+            .min_exclusive(0.0)
+            .required()
+            .build();
+
+        let h_o = ParamBuilder::float(SYMBOLS.h_o)
+            .name("Height of the opening")
+            .units("m")
+            .min_exclusive(0.0)
+            .required()
+            .build();
+
+        let w = ParamBuilder::float(SYMBOLS.w)
+            .name("Width of the wall containing the opening")
+            .units("m")
+            .min_exclusive(0.0)
+            .required()
+            .build();
+
+        let d = ParamBuilder::float(SYMBOLS.d)
+            .name("Depth of the room behind the opening")
+            .units("m")
+            .min_exclusive(0.0)
+            .required()
+            .build();
+
+        params.add(r);
+        params.add(a_t);
+        params.add(a_o);
+        params.add(h_o);
+        params.add(w);
+        params.add(d);
+
+        return params;
+    }
+
+    fn calc_sheet(
+        &self,
+        params: &Parameters,
+        stale: Option<bool>,
+    ) -> framework::method::calculation::ArcCalculation {
+        let r = params.get(SYMBOLS.r);
+        let a_t = params.get(SYMBOLS.a_t);
+        let a_o = params.get(SYMBOLS.a_o);
+        let h_o = params.get(SYMBOLS.h_o);
+        let w = params.get(SYMBOLS.w);
+        let d = params.get(SYMBOLS.d);
+
+        let stale = stale.unwrap_or(false);
+        let calc_sheet: Arc<RwLock<Calculation>> = Arc::new(RwLock::new(Calculation::new(stale)));
+
+        let step = vec![a_t.clone(), a_o.clone(), h_o.clone(), w.clone(), d.clone()];
+        let mut nomenclature = step.clone();
+        nomenclature.push(r.clone());
+
+        let step = Step {
+            name: "Ventilation-controlled rate of burning | Eq. 6.58".to_string(),
+            nomenclature: nomenclature,
+            input: step.clone().into_iter().map(|p| p.into()).collect(),
+            render: true,
+            process: vec![vec![CalculationComponent::Equation(super::equation(
+                r.symbol(),
+                a_t.symbol(),
+                a_o.symbol(),
+                h_o.symbol(),
+                w.symbol(),
+                d.symbol(),
+            ))]],
```

Comment:
The naming of this is confusing. You define `step` here and then you use it on line 166. But you use it within a block that is defining a new variable also called `step`.

```suggestion
        let input = vec![a_t.clone(), a_o.clone(), h_o.clone(), w.clone(), d.clone()];
        let mut nomenclature = input.clone();
        nomenclature.push(r.clone());

        let step = Step {
            name: "Ventilation-controlled rate of burning | Eq. 6.58".to_string(),
            nomenclature: nomenclature,
            input: input.clone().into_iter().map(|p| p.into()).collect(),
            render: true,
            process: vec![vec![CalculationComponent::Equation(super::equation(
                r.symbol(),
                a_t.symbol(),
                a_o.symbol(),
                h_o.symbol(),
                w.symbol(),
                d.symbol(),
            ))]],
```

Reply:
Addressed.